File: Definitions/Game_Definitions/classes/NetrunnerChair.lua
---@meta
---@diagnostic disable

---@class NetrunnerChair : InteractiveDevice
NetrunnerChair = {}

---@return NetrunnerChair
function NetrunnerChair.new() return end

---@param props table
---@return NetrunnerChair
function NetrunnerChair.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function NetrunnerChair:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function NetrunnerChair:OnTakeControl(ri) return end

---@return EGameplayRole
function NetrunnerChair:DeterminGameplayRole() return end

---@return NetrunnerChairController
function NetrunnerChair:GetController() return end

---@return NetrunnerChairControllerPS
function NetrunnerChair:GetDevicePS() return end

File: DecompiledGameScripts/cyberpunk/devices/utilities/netrunner/netrunnerChair.swift

public class NetrunnerChair extends InteractiveDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as NetrunnerChairController;
  }

  private const func GetController() -> ref<NetrunnerChairController> {
    return this.m_controller as NetrunnerChairController;
  }

  public const func GetDevicePS() -> ref<NetrunnerChairControllerPS> {
    return this.GetController().GetPS();
  }

  public const func DeterminGameplayRole() -> EGameplayRole {
    return EGameplayRole.ExplodeLethal;
  }
}
